packages/react-reconciler/src/ReactFiberCommitWork.new.js

```diff
@@ -3737,7 +3737,40 @@ function disconnectPassiveEffect(finishedWork: Fiber): void {
     }
     case OffscreenComponent: {
       const instance: OffscreenInstance = finishedWork.stateNode;
-      if (instance.visibility & OffscreenPassiveEffectsConnected) {
+      if (
+        enableTransitionTracing &&
+        finishedWork.return !== null &&
+        finishedWork.return.tag === SuspenseComponent
+      ) {
+        const isHidden = finishedWork.memoizedState !== null;
+        if (isHidden && instance.pendingMarkers !== null) {
+          instance.pendingMarkers.forEach(
+            ({transitions, pendingBoundaries, name}) => {
+              if (transitions !== null && pendingBoundaries !== null) {
+                if (pendingBoundaries.has(instance)) {
+                  pendingBoundaries.delete(instance);
+                }
+
+                if (name) {
+                  addMarkerProgressCallbackToPendingTransition(
+                    name,
+                    transitions,
+                    pendingBoundaries,
+                  );
+                } else {
+                  transitions.forEach(transition => {
+                    addTransitionProgressCallbackToPendingTransition(
+                      transition,
+                      pendingBoundaries,
+                    );
+                  });
+                }
+              }
+            },
+          );
+        }
+      } else if (instance.visibility & OffscreenPassiveEffectsConnected) {
```

Comment:
This `else if` doesn't look right. It won't run the code in the next branch to disconnect the passive effects.